client/k8scrdclient/k8scrdclient.go

```diff
@@ -103,3 +108,37 @@ func (c *CRDClient) EnsureDeleted(ctx context.Context, customResource *apiextens
 
 	return nil
 }
+
+// ensureStatusSubresourceCreated ensures if the CRD has a status subresource
+// it is created. This is needed if a previous version of the CRD without the
+// status subresource is present.
+func (c *CRDClient) ensureStatusSubresourceCreated(ctx context.Context, customResource *apiextensionsv1beta1.CustomResourceDefinition, backOff backoff.BackOff) error {
+	if customResource.Spec.Subresources == nil || customResource.Spec.Subresources.Status == nil {
+		// Nothing to do.
+		return nil
+	}
+
+	operation := func() error {
+		manifest, err := c.k8sExtClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(customResource.Name, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if manifest.Spec.Subresources == nil || manifest.Spec.Subresources.Status == nil {
+			customResource.SetResourceVersion(manifest.ResourceVersion)
+			_, err = c.k8sExtClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(customResource)
```

Comment:
I don't follow. Why do we actually need to call Update here?

Reply:
We need to call Update if the CRD exists but it doesn't have the status subresource. The current check just creates the CRD if it doesn't exist.

Without this change the chartstatus resource in chart-operator will fail when it calls `UpdateStatus` for the chartconfig CRs.